### ISSUE TITLE
[Feature Request] Circular padding tensor operation

### ISSUE THREAD

**Issue by gartia** [ggerganov/ggml#635]
Honestly i was trying to implement it myself, but i'm not very knowledgeable on c and the code base for ggml is pretty complicated. 

In pytorch you can change padding types. I don't really have a use for the other ones, but circular padding would be extremely useful!
https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html

Example of circular padding
```
[ [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9] ]
->
[ [9, 7, 8, 9, 7],
  [3, 1, 2, 3, 1],
  [6, 4, 5, 6, 4],
  [9, 7, 8, 9, 7],
  [3, 1, 2, 3, 1] ]

```
